adapters/epics.py

```diff
@@ -68,9 +69,14 @@ def doProcess(self, dt):
 
 
 class EpicsAdapter(Adapter):
-    def run(self, target, bindings, *args, **kwargs):
+    def _parseArguments(self, arguments):
+        parser = ArgumentParser(description="Adapter to expose a device via EPICS")
+        parser.add_argument('-p', '--prefix', help='Prefix to use for all PVs', default='')
+        return parser.parse_args(arguments)
+
+    def run(self, target, bindings, prefix):
```

Comment:
Yeah, we can just remove `doRun` in the base class... I only added it since otherwise pycharm complained it didn't exist. It's smart enough not to complain if you check using `hasattr` though... I'm impressed.

Agreed about `kwargs`, it always makes things a bit awkward. There's one place in `StateMachine` that I ended up using it as well, but I'm not sure if it's possible to get rid of it there.

Having tried it, I don't think I'm a big fan of the result of using the `run`/`doRun` indirection here though...

Before, we had a very simple base class that defined a common interface: a `run` method that takes three named arguments. When implementing an adapter, PyCharm will even fill in the argument names for you when you start writing it. Very straightforward.

Now, the base class defines a public `run` that you're *not* supposed to override, a private `_parseArguments` that you *are* supposed to override, and expects you to implement a `doRun` which it doesn't define. And the signature of `doRun`, including the *names* that the arguments *must* have, depends on the contents of your `_parseArguments`. 

And all we've gained is that we don't have to call `_parseArguments` (or something like it, and assuming we need it) manually, no? Is that really worth it?